tensorflow_datasets/image/cityscapes.py

```diff
@@ -0,0 +1,185 @@
+'''
+
+_DESCRIPTION = '''\
+  Cityscapes is a dataset consisting of diverse urban street scenes and dense pixel-level
+  annotations taken across 50 different cities at varying times of the year. The label
+  annotations span across 30 (34 including ingore indices) distinct classes of objects
+  commonly encountered in driving scenes. Detailed label information may be found here:
+  https://github.com/mcordts/cityscapesScripts/blob/master/cityscapesscripts/helpers/labels.py#L52-L99
+
+  The dataset consists of 5000 annotated image, label pairs at 1024 * 2048 resolution
+  pre-split into training (2975) validation (500) and test (1525) sets.
+
+  Cityscapes also provides an additional dataset (accessible via 'cityscapes_coarse')
+  consisting of coarse labels for training and validation splits as well as 19998 images
+  in a new 'train_extra' split which may be useful for data-heavy applications.
+'''
+
+# TODO add instance ids (might need to import cityScapesScripts)
+
+class CityscapesCore(tfds.core.GeneratorBasedBuilder):
+  '''Base for cityscapes datasets'''
+
+  VERSION = tfds.core.Version('2.0.0')
+
+  def _info(self):
+    return tfds.core.DatasetInfo(
+        builder=self,
+        description=(_DESCRIPTION),
+        features=tfds.features.FeaturesDict({
+            'image': tfds.features.Image(shape=(1024, 2048, 3), encoding_format='png'),
+            'label': tfds.features.Image(shape=(1024, 2048, 1), encoding_format='png'),
+            'image_id': tfds.features.Text(),
+        }),
+        supervised_keys=('image', 'label'),
+        urls=["https://www.cityscapes-dataset.com", "https://github.com/mcordts/cityscapesScripts"],
+        citation=_CITATION,
+    )
+
+  def _split_generators(self, dl_manager):
+    raise NotImplementedError
+
+  def _generate_examples(self, images_path, labels_path):
+    for city_id in tf.io.gfile.listdir(images_path):
+      city_images_path = os.path.join(images_path, city_id)
+      city_labels_path = os.path.join(labels_path, city_id)
+
+      for image_file in tf.io.gfile.listdir(city_images_path):
+        image_id = _image_id(image_file)
+        image_path = os.path.join(city_images_path, image_file)
+        label_path = os.path.join(
+            city_labels_path, f'{image_id}_{self.__class__.CITYSCAPES_LABEL_SUFFIX}.png')
+
+        yield {
+            'image': image_path,
+            'label': label_path,
+            'image_id': image_id
+        }
+
+
+class Cityscapes(CityscapesCore):
+  '''Cityscapes with fine labels.'''
+
+  CITYSCAPES_LABEL_SUFFIX = 'gtFine_labelIds'
+
+  def _split_generators(self, dl_manager):
+    paths = {
+        'images': os.path.join(dl_manager.manual_dir, 'leftImg8bit_trainvaltest.zip'),
+        'labels': os.path.join(dl_manager.manual_dir, 'gtFine_trainvaltest.zip'),
+    }
+
+    if any(not os.path.exists(p) for p in paths.values()):
+      msg = 'You must download the dataset files manually and place them in: '
+      msg += ', '.join(paths.values())
```

Comment:
can you add some explanation on why the tfds cannot download the files?

Reply:
Cityscapes (much like ImageNet) requires a login before you can download the files.

Reply:
:-(  I see. please mention it in the comment here.